monai/apps/deepgrow/transforms.py

```diff
@@ -0,0 +1,623 @@
+            if neg:
+                guidance[0].append([-1] * len(neg))
+                guidance[1].append(neg)
+        else:
+            for g, d, p in zip(guidance, discrepancy, probability):
+                pos, neg = self.add_guidance(d, p)
+                if pos:
+                    g[0].append(pos)
+                    g[1].append([-1] * len(pos))
+                if neg:
+                    g[0].append([-1] * len(neg))
+                    g[1].append(neg)
+        return np.asarray(guidance)
+
+    def __call__(self, data):
+        guidance = data[self.guidance]
+        discrepancy = data[self.discrepancy]
+        probability = data[self.probability]
+
+        data[self.guidance] = self._apply(guidance, discrepancy, probability)
+        return data
+
+
+class SpatialCropForegroundd(MapTransform):
+    def __init__(
+        self,
+        keys,
+        source_key: str,
+        spatial_size,
+        select_fn=lambda x: x > 0,
+        channel_indices=None,
+        margin: int = 0,
+        meta_key_postfix="meta_dict",
+        start_coord_key: str = "foreground_start_coord",
+        end_coord_key: str = "foreground_end_coord",
+        original_shape_key: str = "foreground_original_shape",
+        cropped_shape_key: str = "foreground_cropped_shape",
+    ) -> None:
+        super().__init__(keys)
+
+        self.source_key = source_key
+        self.spatial_size = list(spatial_size)
+        self.select_fn = select_fn
+        self.channel_indices = channel_indices
+        self.margin = margin
+        self.meta_key_postfix = meta_key_postfix
+        self.start_coord_key = start_coord_key
+        self.end_coord_key = end_coord_key
+        self.original_shape_key = original_shape_key
+        self.cropped_shape_key = cropped_shape_key
+
+    def __call__(self, data):
+        box_start, box_end = generate_spatial_bounding_box(
+            data[self.source_key], self.select_fn, self.channel_indices, self.margin
+        )
+
+        center = np.mean([box_start, box_end], axis=0).astype(int).tolist()
+        current_size = np.subtract(box_end, box_start).astype(int).tolist()
+
+        if np.all(np.less(current_size, self.spatial_size)):
+            cropper = SpatialCrop(roi_center=center, roi_size=self.spatial_size)
+            box_start = cropper.roi_start
+            box_end = cropper.roi_end
+        else:
+            cropper = SpatialCrop(roi_start=box_start, roi_end=box_end)
+
+        for key in self.keys:
+            meta_key = f"{key}_{self.meta_key_postfix}"
+            data[meta_key][self.start_coord_key] = box_start
+            data[meta_key][self.end_coord_key] = box_end
+            data[meta_key][self.original_shape_key] = data[key].shape
+
+            image = cropper(data[key])
+            data[meta_key][self.cropped_shape_key] = image.shape
+            data[key] = image
+        return data
+
+
+# Transforms to support Inference
+class SpatialCropGuidanced(MapTransform):
```

Comment:
This name kind of suggest this transform is doing spatial cropping on the guidance?
But this transform is actually doing spatial cropping on the images (specified by keys)
and update the pos/neg guidances' coordinates accordingly

Should we come up with another name?

Reply:
it's doing based on guidance.. the bounding box is defined by user clicks/guidance... added the docs to say that